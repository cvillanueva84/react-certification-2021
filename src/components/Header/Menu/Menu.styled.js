import styled from 'styled-components';

export const HamburgerMenu = styled.div`
  display: block;
  position: relative;
  top: 2px;
  left: 0px;

  z-index: 1;

  -webkit-user-select: none;
  user-select: none;
  height: 100%;

  a {
    text-decoration: none;
    color: #232323;

    transition: color 0.3s ease;
    padding: 0;
    text-align: center;
    margin-top: 20px;
  }

  a:hover {
    color: white;
  }
`;

export const SpanHamburguer = styled.span`
  display: block;
  width: 33px;
  height: 4px;
  margin-bottom: 5px;
  position: relative;

  background: #ffffff;
  border-radius: 3px;

  z-index: 1;

  transform-origin: 4px 0px;

  transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1),
    background 0.5s cubic-bezier(0.77, 0.2, 0.05, 1), opacity 0.55s ease;

  &:first-child {
    transform-origin: 0% 0%;
  }

  &:nth-last-child(2) {
    transform-origin: 0% 100%;
  }
`;

export const Menu = styled.ul`
  position: fixed;
  top: 10px;
  width: 300px;
  margin: -10px 0 0 -50px;
  padding: 50px;
  padding-top: 45px;
  height: 100vh;

  background: #ededed;
  list-style-type: none;
  -webkit-font-smoothing: antialiased;

  transform-origin: 0% 0%;
  transform: translate(-100%, 0);

  transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1);

  li {
    padding: 10px 0;
    font-size: 22px;
  }
`;

export const MenuCover = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  inset: 0;
  background-color: rgba(0, 0, 0, 0.7);
  width: 100vw;
  height: 100vh;
  display: none;
  overflow: hidden;
  transition: display 0.5s cubic-bezier(0.77, 0.2, 0.05, 1);
`;

export const MenuButton = styled.input`
  display: block;
  width: 40px;
  height: 32px;
  position: absolute;
  cursor: pointer;
  top: -5px;
  left: -5px;
  opacity: 0; /* hide this */
  z-index: 2; /* and place it over the hamburger */
  -webkit-touch-callout: none;

  &:checked ~ ${SpanHamburguer} {
    opacity: 1;
    transform: rotate(45deg) translate(-2px, -1px);
    background: #232323;
  }

  &:checked ~ ${SpanHamburguer}:nth-last-child(3) {
    opacity: 0;
    transform: rotate(0deg) scale(0.2, 0.2);
  }

  &:checked ~ ${SpanHamburguer}:nth-last-child(2) {
    transform: rotate(-45deg) translate(0, -1px);
  }

  &:checked ~ ${Menu} {
    transform: none;
  }

  &:checked ~ ${MenuCover} {
    display: block;
  }
`;

export const ContainerHamburguer = styled.div`
  display: 'flex';
  flex-direction: 'column';
`;
