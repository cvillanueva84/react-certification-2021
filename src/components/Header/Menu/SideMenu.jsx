import React, { useRef } from 'react';
import { Link } from 'react-router-dom';
import {
  HamburgerMenu,
  Menu,
  MenuButton,
  MenuCover,
  SpanHamburguer,
} from './Menu.styled';

const SideMenu = () => {
  const ref = useRef(null);

  const hidenMenu = () => {
    ref.current.checked = false;
  };

  return (
    <HamburgerMenu>
      <MenuButton ref={ref} type="checkbox" data-testid="hamburguerMenu" />
      <MenuCover onClick={hidenMenu} />
      <SpanHamburguer />
      <SpanHamburguer />
      <SpanHamburguer />

      <Menu>
        <Link to="/" onClick={hidenMenu}>
          <li>Home</li>
        </Link>
        <Link to="/" onClick={hidenMenu}>
          <li>Home</li>
        </Link>
      </Menu>
    </HamburgerMenu>
  );
};

export default SideMenu;
