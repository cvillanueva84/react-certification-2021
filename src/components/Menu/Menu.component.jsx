import React, { useContext } from 'react';
import styled from 'styled-components';
import { useHistory } from 'react-router-dom';

import { StateContext } from '../../context/State/state';

const StyledMenu = styled.nav`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  border: 3px solid ${(props) => (props.mode ? '#ffffff' : '#303030')};
  height: 100vh;
  text-align: left;
  padding: 2rem;
  position: absolute;
  top: 0;
  left: 0;
  transition: transform 0.3s ease-in-out;
  transform: translateX(-100%);
  transform: ${({ open }) => (open ? 'translateX(0)' : 'translateX(-100%)')};
  z-index: 2;
  background-color: ${(props) => (props.mode ? '#303030' : '#ffffff')};
  color: ${(props) => (props.mode ? '#ccc' : '#303030')};
`;

const Span = styled.span`
  cursor: pointer;
`;

const Menu = ({ open }) => {
  const stateContext = useContext(StateContext);
  const history = useHistory();
  const {
    darkMode,
    video: { openedDetails },
    toggleOpenedDetails,
  } = stateContext;

  return (
    <StyledMenu mode={darkMode ? 1 : 0} open={open}>
      <Span
        onClick={() => {
          history.push('/');
          if (openedDetails) {
            toggleOpenedDetails(openedDetails);
          }
        }}
      >
        HOME
      </Span>
      <Span
        onClick={() => {
          history.push('/favorites');
          if (openedDetails) {
            toggleOpenedDetails(openedDetails);
          }
        }}
      >
        FAVORITES
      </Span>
    </StyledMenu>
  );
};

export default Menu;
