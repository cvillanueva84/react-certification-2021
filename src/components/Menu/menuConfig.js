export const listMenu = [
    { route: '/', menu: 'Home', icon: 'home' },
    { route: '/favorites', menu: 'Favorites', icon: 'favorite' },
];
