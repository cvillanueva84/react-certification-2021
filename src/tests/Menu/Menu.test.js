import React from 'react';
import { shallow } from 'enzyme';
import { useState } from 'react';
import { Menu } from '../../components/Menu/Menu.component';

describe('Menu', () => {
    test('should render correctly', () => {
        const setIsOpen = jest.fn();
        const wrapper = shallow(<Menu isOpen={false} setIsOpen={setIsOpen} /> );
        expect(wrapper).toMatchSnapshot();
    });
});
