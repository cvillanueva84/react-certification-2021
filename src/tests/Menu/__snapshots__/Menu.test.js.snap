// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  onClose={[Function]}
  open={false}
>
  <div
    role="presentation"
  >
    <WithStyles(ForwardRef(List))>
      <WithStyles(ForwardRef(ListItem))
        button={true}
        key="Home"
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(Icon))>
            home
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Home"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))
        button={true}
        key="Favorites"
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(Icon))>
            favorite
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Favorites"
        />
      </WithStyles(ForwardRef(ListItem))>
    </WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(Divider)) />
  </div>
</WithStyles(ForwardRef(Drawer))>
`;
